src/main/kotlin/se/kth/spork/spoon/pcsinterpreter/SporkTreeBuilder.kt

```diff
@@ -263,25 +279,15 @@ internal class SporkTreeBuilder(
     private fun extractConflictList(
         pcs: Pcs<SpoonNode>,
         siblings: Map<SpoonNode, Pcs<SpoonNode>>,
+        baseSiblings: Map<SpoonNode, Pcs<SpoonNode>>?,
     ): List<SpoonNode> {
         var currentPcs = pcs
         val nodes: MutableList<SpoonNode> = mutableListOf()
         while (true) {
-            val conflicts = delta.structuralConflicts[currentPcs]
-            if (conflicts != null && conflicts.isNotEmpty()) {
-                val finalPcs = currentPcs
-                val predConflict = conflicts.stream()
-                    .filter {
-                        isPredecessorConflict(
-                            finalPcs,
-                            it,
-                        )
-                    }
-                    .findFirst()
-                if (predConflict.isPresent) {
-                    remainingInconsistencies.remove(predConflict.get())
-                    return nodes
-                }
+            val otherPred = delta.getOtherPredecessors(currentPcs).firstOrNull()
+            if (otherPred != null) {
+                remainingInconsistencies.remove(otherPred)
+                return nodes
```

Comment:
This is quite a simplification from the original so I suspect I might have overseen something here. Curious to know what you think.